core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -753,8 +753,9 @@ public void testListNamespace() {
     TableIdentifier tbl4 = TableIdentifier.of("db", "metadata");
     TableIdentifier tbl5 = TableIdentifier.of("db2", "metadata");
     TableIdentifier tbl6 = TableIdentifier.of("tbl6");
+    TableIdentifier tbl7 = TableIdentifier.of("db2", "ns4", "tbl5");
```

Comment:
How does this test change exercise the problem?
this new additional doesn't include `_` nor `%`

Reply:
This namespace was returned for `listNamespaces(Namespace.of("db"))` because of the following condition:
https://github.com/apache/iceberg/blob/24d26b6a35a1287531e72357691d9dbd3d7f79bd/core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java#L447

Anyway, I added two namespaces having `_` or `%`.